src/helpers/mergeContact.js

```diff
@@ -0,0 +1,55 @@
+import mergeWith from 'lodash/mergeWith'
+import merge from 'lodash/merge'
+import concat from 'lodash/concat'
+import groupBy from 'lodash/groupBy'
+import map from 'lodash/map'
+
+export function mergeContact(contact, attributes) {
+  return mergeWith({}, contact, attributes, customizer)
+}
+
+const strategies = {
+  email: mergeArrayOfObject('address'),
+  phone: mergeArrayOfObject('number'),
+  givenName: keepLonger,
+  familyName: keepLonger
+}
+
+const fromStrategies = strategies => (objValue, srcValue, key) =>
+  strategies[key] ? strategies[key](objValue, srcValue, key) : undefined
+
+const customizer = fromStrategies(strategies)
+
+function mergeArrayOfObject(key) {
+  return (objValue, srcValue) => {
+    return mergeArrayOnKey(objValue, srcValue, key)
+  }
+}
+
+function keepLonger(objValue = '', srcValue = '') {
+  return objValue.length > srcValue.length ? objValue : srcValue
```

Comment:
We use the length of the string as a basis for the strategy that wants to keep the longest string.

The Product Owner wrote a story that says the user wants to keep the longest chain because it is a good heuristic to keep the richest information.
Unless unrealistic examples such as `"John''` vs `"AAAAAAAAAAAAA''`, we can consider that `"John''` contains less information than `"Lord Commander John''` or `"Jonathan''`.